sympy/ntheory/partitions_.py

```diff
@@ -172,7 +171,7 @@ def npartitions(n, verbose=False):
         math.log(10, 2)
     prec = p = int(pbits*1.1 + 100)
     s = fzero
-    M = max(6, int(0.24*n**0.5 + 4))
+    M = math.ceil(n**0.5)
```

Comment:
This change doesn't seem to match the description in the OP "Takes more terms in the series approximation and increases the arithmetic precision." although to be clear I haven't read the issue in detail and don't know exactly what this does

Reply:
This change takes more terms in the series approximation.

Reply:
I'm not going to read references etc so I'll leave this for someone else to review.

What I will say though is that I would use an integer root function rather than floating point here.